x/sync/db.go

```diff
@@ -12,3 +19,16 @@ type DB interface {
 	merkledb.ChangeProofer
 	merkledb.RangeProofer
 }
+
+type ProofClient interface {
+	merkledb.Clearer
+	merkledb.MerkleRootGetter
```

Comment:
nit: naming: These are no longer merkledb interfaces, correct? Sure, they are implemented by merkledb but IIRC golang just lets you declare interfaces. If so, interfaces used by ProofClient (or SyncClient) should be maybe in a package named xsync or something.